### PR TITLE
feat(blog): Add gatsby-plugin-transition-link blog post

### PR DESCRIPTION
**Pull request by TylerBarnes**
As requested by @jlengstorf in #5213 this is a gatsby blog post about the page transition plugin I built.

### CODE REVIEW

File: docs/blog/2018-12-04-per-link-gatsby-page-transitions-with-transitionlink/index.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+TransitionPortal is a React portal which allows you to position animation elements above both your exiting and entering pages. Sometimes you'll need to have something persist throughout the whole animation and some other page element will overlap it. Portals are great for solving that.
+
+[Check the docs](https://transitionlink.tylerbarnes.ca/docs) for more info on TransitionState and TransitionPortal.
+
+### Capping it off
+
+I'm excited to see what you and the rest of the Gatsby community will build with TransitionLink! Feel free to send any unanswered questions or issues you may have [my way](https://github.com/TylerBarnes/gatsby-plugin-transition-link/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc). I'll do my best to help you out asap.
+
+— Ty. B.
```
Comment (jlengstorf) [inline docs/blog/2018-12-04-per-link-gatsby-page-transitions-with-transitionlink/index.md]: We tend to avoid signatures on blog posts for consistency.

Comment (TylerBarnes) [reply docs/blog/2018-12-04-per-link-gatsby-page-transitions-with-transitionlink/index.md]: That makes sense. I removed it

**Comment by TylerBarnes** [comment]
Awesome, thanks for the feedback @jlengstorf! Definitely good suggestions.

**Comment by TylerBarnes** [comment]
Awesome @m-allanson! Thanks so much :) I'm stoked I could contribute to the gatsby blog. When you get a chance to try it out I'd love to hear your thoughts!

**Pull request merged**